nav2_behavior_tree/plugins/condition/ml_vicinity_condition.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2024 Andy Zelenak
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_behavior_tree/plugins/condition/ml_vicinity_condition.hpp"
+#include <chrono>
+#include <cstdlib>
+#include <curl/curl.h>
+#include <nlohmann/json.hpp>
+#include <string>
+
+namespace nav2_behavior_tree
+{
+
+MLVicinityCondition::MLVicinityCondition(
+  const std::string & condition_name,
+  const BT::NodeConfiguration & conf)
+: BT::ConditionNode(condition_name, conf)
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+
+  server_timeout_ = config().blackboard->template get<std::chrono::milliseconds>("server_timeout");
+  getInput<std::chrono::milliseconds>("server_timeout", server_timeout_);
+  std::string image_topic;
+  getInput<std::string>("image_topic", image_topic);
+  image_sub_ = node_->create_subscription<sensor_msgs::msg::Image>(
+      image_topic, 10, std::bind(&MLVicinityCondition::imageCallback, this, std::placeholders::_1));
+}
+
+BT::NodeStatus MLVicinityCondition::tick()
+{
+  if (promptAIModel())
+  {
+    return BT::NodeStatus::SUCCESS;
+  }
+  return BT::NodeStatus::FAILURE;
+}
+
+bool MLVicinityCondition::promptAIModel()
```

Comment:
i figured i'd cram everything related to ChatGPT into this one function so it can be swapped out easily, although it's not the most efficient way